aten/src/ATen/cpu/vec256/intrinsics.h

```diff
@@ -27,6 +27,30 @@
         (defined(__VEC__) || defined(__ALTIVEC__))
 /* XLC or GCC-compatible compiler, targeting PowerPC with VMX/VSX */
 #include <altivec.h>
+#include <altivec.h>
+#define offset0   0
```

Comment:
Macros in a top level header is a bad taste (difficult to control for namespace collisions); I'm not sure what you are using these for but consider using constexpr.

Reply:
yes, it should be removed.  offsets I am using for explicit load and stores, but using it as C style constant is not right here as you mentioned.
In reality, GCC compilers are capable of translating the subscript style` vec[index]` memory operations, but some versions had problems choosing proper instructions. I mostly chose to use explicit vector intrinsics names